akka-actor/src/main/scala/akka/actor/dungeon/ReceiveTimeout.scala

```diff
@@ -28,16 +28,13 @@ private[akka] trait ReceiveTimeout { this: ActorCell ⇒
 
   final def checkReceiveTimeout() {
     val recvtimeout = receiveTimeoutData
-    //Only reschedule if desired and there are currently no more messages to be processed
-    if (!mailbox.hasMessages) recvtimeout._1 match {
+    recvtimeout._1 match {
```

Comment:
I agree that this is the most straight forward way to implement it. I'm somewhat concerned about the performance concern of always rescheduling, but other solutions I can think of are too complicated and have other drawbacks.